### PR TITLE
SNOW-1043081: Adding support for qualified names for image repositories.

### PR DESCRIPTION
**Pull request by sfc-gh-davwang**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Modified implementation of show_specific_object to allow qualified object names and added support for qualified image repository names by removing validation that image repository names are unqualified. Also added an explicit check that show_specific_object only returns a single row and throws an error otherwise.

**Usage**:
image

**Errors when database, schema, repository don't exist**:
image

### CODE REVIEW

File: tests/spcs/test_image_repository.py [hunk 1]
```diff
@@ -229,46 +218,22 @@ def test_get_repository_url_no_scheme(mock_get_row, mock_check_database_and_sche
...
     mock_conn.database = "DB"
     mock_conn.schema = "SCHEMA"
-    with pytest.raises(ClickException) as e:
+    with pytest.raises(ProgrammingError) as e:
         ImageRepositoryManager().get_repository_url(repo_name="IMAGES")
     assert (
-        e.value.message
-        == "Image repository 'IMAGES' does not exist in database 'DB' and schema 'SCHEMA' or not authorized."
+        e.value.msg
+        == f"Image repository 'DB.SCHEMA.IMAGES' does not exist or not authorized."
```
Comment (sfc-gh-yiwang) [inline tests/spcs/test_image_repository.py:235]: nit: f is not needed

File: tests/spcs/test_image_repository.py [hunk 2]
```diff
@@ -229,46 +218,22 @@ def test_get_repository_url_no_scheme(mock_get_row, mock_check_database_and_sche
...
     )
     mock_get_row.assert_called_once_with(
         "image repositories", "IMAGES", check_schema=True
     )
-
-
-@mock.patch(
-    "snowflake.cli.plugins.spcs.image_repository.manager.ImageRepositoryManager._conn"
-)
-def test_get_repository_url_no_database(mock_conn):
```
Comment (sfc-gh-yiwang) [inline tests/spcs/test_image_repository.py:261]: Why do we remove this test?

**Review by sfc-gh-yiwang — DISMISSED** [review]
Overall LGTM

File: tests/spcs/test_image_repository.py (hunk 2 above)
Comment (sfc-gh-davwang) [reply tests/spcs/test_image_repository.py:261]: added back

**Review by sfc-gh-yiwang — APPROVED** [review]

File: src/snowflake/cli/api/sql_execution.py [hunk 3]
```diff
@@ -82,44 +82,27 @@ def use_role(self, new_role: str):
...
+            _, schema, database = from_qualified_name(name)
+        else:
+            schema, database = None, None
+        schema = schema or self._conn.schema
+        database = database or self._conn.database
         if not database:
             raise DatabaseNotProvidedError()
-        try:
-            self._execute_query(f"USE DATABASE {database}")
-        except ProgrammingError as e:
```
Comment (sfc-gh-davwang) [inline src/snowflake/cli/api/sql_execution.py:106]: Note that I removed the execution of USE DATABASE and USE SCHEMA in use_schema_query. I think that it makes sense to check a database and schema are provided but if they are necessary for the query then an error will be raised when executing the main query if the schema or database do not actually exist.

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**